public/components/CampaignNotes/CampaignNotes.js

```diff
@@ -0,0 +1,61 @@
+import React, {Component, PropTypes} from 'react';
+import CampaignNotesList from '../CampaignNotesList/CampaignNotesList';
+import CampaignNotesAdd from './CampaignNotesAdd';
+
+class CampaignNotes extends Component {
+
+  componentWillMount() {
+    this.props.campaignNoteActions.getCampaignNotes(this.props.campaign.id);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.campaign.id !== this.props.campaign.id) {
+      this.props.campaignNoteActions.getCampaignNotes(nextProps.campaign.id);
+    }
+
+  }
+
+  saveNote = () => {
+      this.props.campaignNotesAddActions.createNote(this.props.id, { content: this.state.message });
+      window.removeEventListener('click', this.disableAdding, true);
```

Comment:
Yeah the removal of the EventListener should really be happening in the same component as it's being added (So in the future if this component is replaced then the new one wouldn't need to also do this). It will mean that you can't directly pass down the createNote function, but I think t's worth it